cookbooks/bach_krb5/libraries/utils.rb

```diff
@@ -0,0 +1,18 @@
+def principal_exists?(pname) 
+  require 'rkerberos'
+  kadm5_connected = false
+  princ_found = false
+
+  begin
+    kadm5 = Kerberos::Kadm5.new(:principal => "#{node[:krb5][:admin_principal]}", :password => "#{get_config("krb5-admin-password")}")
+    kadm5_connected = true
+  rescue Kerberos::Kadm5::Exception
```

Comment:
I was mainly wondering. However, I really would rather us `raise` or if the error message provided by the Kerberos code is good enough, let it raise for us. (See [What Makes A Good Chef Cookbook?](http://www.slideshare.net/JulianDunn/what-makes-a-good-cookbook-35029540) for recommendations on how not to use `Chef::Application.fatal` -- a [not recommended](https://tickets.opscode.com/browse/CHEF-4982) API to use.)